go-controller/pkg/ovn/egressgw.go

```diff
@@ -569,41 +569,32 @@ func (oc *Controller) delHybridRoutePolicyForPod(podIP net.IP, node string, forc
 // cleanUpBFDEntry checks if the BFD table entry related to the associated
 // gw router / port / gateway ip is referenced by other routing rules, and if
 // not removes the entry to avoid having dangling BFD entries.
-// This is temporary and can be safely removed when we consume an ovn version
-// that includes http://patchwork.ozlabs.org/project/ovn/patch/3c39dc96a36a3445cfa8485a67de79f9f3d5651b.1614602770.git.lorenzo.bianconi@redhat.com/
-func cleanUpBFDEntry(gatewayIP, gatewayRouter, prefix string) {
+func (oc *Controller) cleanUpBFDEntry(gatewayIP, gatewayRouter, prefix string) {
 	portName := prefix + types.GWRouterToExtSwitchPrefix + gatewayRouter
 
-	output, stderr, err := util.RunOVNNbctl(
-		"--format=csv", "--data=bare", "--no-heading", "--columns=bfd", "find", "Logical_Router_Static_Route", "output_port="+portName, "nexthop=\""+gatewayIP+"\"", "bfd!=[]")
-
-	if err != nil {
-		klog.Errorf("cleanUpBFDEntry: failed to list routes for %s, stderr: %q, (%v)", portName, gatewayIP, err, stderr)
-		return
-	}
-	// the bfd entry is still referenced, meaning there's another route on the router
-	// referencing it.
-	if strings.TrimSpace(output) != "" {
-		return
-	}
-	uuids, stderr, err := util.RunOVNNbctl(
-		"--format=csv", "--data=bare", "--no-heading", "--columns=_uuid", "find", "BFD", "logical_port="+portName, "dst_ip=\""+gatewayIP+"\"")
+	logicalRouterStaticRouteRes := []nbdb.LogicalRouterStaticRoute{}
+	err := oc.nbClient.WhereCache(func(lrsr *nbdb.LogicalRouterStaticRoute) bool {
+		return lrsr.OutputPort != nil && *lrsr.OutputPort == portName && lrsr.Nexthop == gatewayIP && lrsr.BFD != nil && *lrsr.BFD != ""
+	}).List(&logicalRouterStaticRouteRes)
```

Comment:
why are we checking for `lrsr.OutputPort != nil && lrsr.OutputPort == portName` ?

Shouldn't `lrsr.OutputPort == portName` be enough? `portName` here (prefix + types.GWRouterToExtSwitchPrefix + gatewayRouter) won't be nil.

Reply:
> Shouldn't lrsr.OutputPort == portName be enough?

No, the `Outport` is not a string, it's a pointer to a string. Though the `portName` will not be nil, `Output` can be. If it's nil: we will panic, see the following example: https://play.golang.org/p/6kwF3mTkvra

Reply:
interesting, looks like this is something we'd need to do for a lot of fields.